src/redux/actions/postingActions.js

```diff
@@ -3,36 +3,188 @@ import {
   DELETE_POSTING,
   LOAD_ITEM_DETAIL,
   UPDATE_ITEM_DETAIL,
+  LOAD_POSTINGS,
+  ERROR_SNACKBAR,
 } from "../constants/actionTypes";
+import { setLoading } from "./loadingActions";
+import axios from "axios";
 
-export const addPosting = (posting, currentUser) => {
+const BASE_URL = `${process.env.REACT_APP_BASE_URL}`;
+
+const addPostingSuccess = (posting) => {
   return {
     type: ADD_POSTING,
     posting: posting,
-    currentUser: currentUser,
   };
 };
 
-export const loadItemDetail = (itemId, postings) => {
+const addPostingError = () => {
   return {
-    type: LOAD_ITEM_DETAIL,
-    itemId: itemId,
-    // TODO: postings unnecessary for when BE integrated
+    type: ERROR_SNACKBAR,
+    msg: "Something went wrong. Posting was not created",
+  };
+};
+
+const loadAllPostingsSuccess = (postings) => {
+  return {
+    type: LOAD_POSTINGS,
     postings: postings,
   };
 };
 
-export const updateItemDetail = (itemId, details) => {
+const loadPostingsError = () => {
+  return {
+    type: ERROR_SNACKBAR,
+    msg: "Something went wrong. Postings could not be loaded",
+  };
+};
+
+const loadItemDetailSuccess = (item) => {
+  return {
+    type: LOAD_ITEM_DETAIL,
+    item: item,
+  };
+};
+
+const loadItemDetailError = (msg) => {
+  return {
+    type: ERROR_SNACKBAR,
+    msg: msg,
+  };
+};
+
+const updateItemDetailSuccess = (itemId, details) => {
   return {
     type: UPDATE_ITEM_DETAIL,
     itemId: itemId,
     details: details,
   };
 };
 
-export const deletePosting = (itemId) => {
+const updateItemDetailError = () => {
+  return {
+    type: ERROR_SNACKBAR,
+    msg: "Something went wrong. Posting was not updated",
+  };
+};
+
+const deletePostingSuccess = (itemId) => {
   return {
     type: DELETE_POSTING,
     itemId: itemId,
   };
 };
+
+const deletePostingError = () => {
```

Comment:
same thing here. A lot of these ERROR actions are doing the same thing. You could simply pass the specific message into the same action from where you dispatch them.